test/test_integration_cluster.rb

```diff
@@ -226,7 +226,10 @@ def test_idle_timeout
 
     get_worker_pids # wait for workers to boot
 
-    connect
+    10.times {
+      fast_connect
+      sleep 0.5
+    }
```

Comment:
The updated test fails on master with:

![Screenshot 2023-12-01 at 9 56 34 am](https://github.com/puma/puma/assets/54629302/cd6748ec-a1b0-4bc5-9a4d-c3ea07167d06)